static/app/components/onboarding/gettingStartedDoc/utils/useCurrentProjectState.tsx

```diff
@@ -1,24 +1,40 @@
-import {useEffect, useState} from 'react';
+import {useEffect, useMemo, useState} from 'react';
+import partition from 'lodash/partition';
 
-import {SidebarPanelKey} from 'sentry/components/sidebar/types';
-import {feedbackOnboardingPlatforms} from 'sentry/data/platformCategories';
+import type {SidebarPanelKey} from 'sentry/components/sidebar/types';
 import PageFiltersStore from 'sentry/stores/pageFiltersStore';
 import {useLegacyStore} from 'sentry/stores/useLegacyStore';
-import type {Project} from 'sentry/types';
+import type {PlatformKey, Project} from 'sentry/types';
 import useProjects from 'sentry/utils/useProjects';
 
-function useCurrentProjectState({currentPanel}: {currentPanel: '' | SidebarPanelKey}) {
+type Props = {
+  allPlatforms: readonly PlatformKey[];
+  currentPanel: '' | SidebarPanelKey;
+  onboardingPlatforms: readonly PlatformKey[];
+  targetPanel: SidebarPanelKey;
+};
+
+function useCurrentProjectState({
+  currentPanel,
+  targetPanel,
+  onboardingPlatforms,
+  allPlatforms,
+}: Props) {
   const [currentProject, setCurrentProject] = useState<Project | undefined>(undefined);
   const {projects, initiallyLoaded: projectsLoaded} = useProjects();
   const {selection, isReady} = useLegacyStore(PageFiltersStore);
 
-  const isActive = currentPanel === SidebarPanelKey.FEEDBACK_ONBOARDING;
+  const isActive = currentPanel === targetPanel;
 
   // Projects with onboarding instructions
   const projectsWithOnboarding = projects.filter(
-    p => p.platform && feedbackOnboardingPlatforms.includes(p.platform)
+    p => p.platform && onboardingPlatforms.includes(p.platform)
   );
 
+  const [supportedProjects, unsupportedProjects] = useMemo(() => {
```

Comment:
feedback doesn't need this, but replay & DDM both do, so adding it into the new refactored hook